src/ucp/core/ucp_ep.c

```diff
@@ -544,6 +544,20 @@ void ucp_ep_destroy_internal(ucp_ep_h ep, const char *message)
 
 static void ucp_ep_disconnected(ucp_ep_h ep)
 {
+    ucp_recv_desc_t  *rdesc;
+
+    while (!ucs_queue_is_empty(&ep->stream_data)) {
```

Comment:
let's extract it to a function - e.g ucp_ep_stream_data_purge() or similar

Reply:
In next PRs i'll extract stream_data and other stream specific stuff to special EP extension which is allocated/initialized and freed in corresponding functions only when this CAP requested. The data will be purged there if you dont have objection to keep it as is for now.

Reply:
ok